### PR TITLE
GH-39712: [Java] Enable code review and formatting code through Spotless Maven plugin

### LINKED ISSUE #39712: [Java] Enable code review and formatting code through Spotless Maven plugin
### Describe the enhancement requested

As Java code increases, it will become increasingly important to implement code formatter best practices. As a result, we will be able to use [spotless](https://github.com/diffplug/spotless) to:

- Reviewing code: Spotless can be used to identify issues related to format.
- Formatting code: Spotless can be used to automatically fix common formatting mistakes, such as spaces, newlines, and unnecessary imports.

This enhancement will propose replacing `maven-checkstyle-plugin` with `spotless-maven-plugin`, which is currently used by Java.

### Component(s)

Java

### PR DESCRIPTION
**Pull request by davisusanibar**
### Rationale for this change

To enable code review and formatting code by using [Spotless Maven plugin](https://github.com/diffplug/spotless).

### What changes are included in this PR?

- [X] Maven configuration to enable Spotless Maven plugin.
- [x] Documentation how to use ´mvn spotless:apply´

### Are these changes tested?

By ´mvn spotless:apply´

### Are there any user-facing changes?

No
* Closes: #39712

### CODE REVIEW

**Comment by davisusanibar** [comment]
Please consider: Initial formatting implementations will require a large number of file changes.

Among the main changes that need to be reviewed are:
- pom.xml
- maven/pom.xml
- bom/pom.xml

**Comment by vibhatha** [comment]
@davisusanibar shall we hold this merge until we merge the errorprone fixes? 
Then it will help a lot to clean up everything at once?

**Comment by davisusanibar** [comment]
> @davisusanibar shall we hold this merge until we merge the errorprone fixes? Then it will help a lot to clean up everything at once?

Hi @vibhatha, this sounds like a good idea to me.

**Comment by davisusanibar** [comment]
The documentation has been added, please review if more details need to be added.

CC: @vibhatha @lidavidm @danepitkin

**Comment by davisusanibar** [comment]
> The documentation has been added, please review if more details need to be added.
> 
> CC: @vibhatha @lidavidm @danepitkin

File at: https://github.com/apache/arrow/blob/b6d012523c3d1627851bde20944401114259d9c6/docs/source/developers/java/development.rst#code-style

**Comment by lidavidm** [comment]
(1) did you mean to leave this in draft
(2) is it perhaps possible to enable this one module at a time?

**Comment by davisusanibar** [comment]
> (1) did you mean to leave this in draft (

Our plan was to merge first https://github.com/apache/arrow/pull/39777 https://github.com/apache/arrow/pull/39713#issuecomment-1903108403

> 2) is it perhaps possible to enable this one module at a time?

Let me break this down into modules.

File: java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrow.java [hunk 1]
```diff
@@ -29,73 +28,58 @@
...
+ * ArrowType.Decimal(precision, scale) DECIMAL --> ArrowType.Decimal(precision, scale) BIT -->
+ * ArrowType.Bool TINYINT --> ArrowType.Int(8, signed) SMALLINT --> ArrowType.Int(16, signed)
+ * INTEGER --> ArrowType.Int(32, signed) BIGINT --> ArrowType.Int(64, signed) REAL -->
+ * ArrowType.FloatingPoint(FloatingPointPrecision.SINGLE) FLOAT -->
+ * ArrowType.FloatingPoint(FloatingPointPrecision.SINGLE) DOUBLE -->
+ * ArrowType.FloatingPoint(FloatingPointPrecision.DOUBLE) BINARY --> ArrowType.Binary VARBINARY -->
+ * ArrowType.Binary LONGVARBINARY --> ArrowType.Binary DATE --> ArrowType.Date(DateUnit.MILLISECOND)
+ * TIME --> ArrowType.Time(TimeUnit.MILLISECOND, 32) TIMESTAMP -->
+ * ArrowType.Timestamp(TimeUnit.MILLISECOND, timezone=null) CLOB --> ArrowType.Utf8 BLOB -->
+ * ArrowType.Binary
```
Comment (danepitkin) [inline java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrow.java:42]: Here's an example of a change we wouldn't want. There might be a few of these types of issues around.

Comment (lidavidm) [reply java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrow.java:42]: Arguably, the original formatting is wrong (javadoc would have rendered it wrong too) and this should've been an HTML definition list

Comment (lidavidm) [reply java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrow.java:42]: But yes, it'll be easier to review if this is broken down

**Comment by davisusanibar** [comment]
In order to define a better alternative solution, the current pull request has been closed

**Comment by vibhatha** [comment]
> In order to define a better alternative solution, the current pull request has been closed

Created an issue: https://github.com/apache/arrow/issues/40757

**Pull request closed without merging**